Chore/143410115/fend unit test

jm-wanja opened the pull request:
**What does this PR do?**
Setup unit testing for the front end

**Description of Task to be completed?**
To ensure the front end has all functionality work the way it was intended. Use Jasmine and Karma (with PhantomJS/Chrome) to setup front end testing environment.

**What are the relevant pivotal tracker stories?**
#143410115

jm-wanja committed:
chore(testing): setup unit testing for front-end

- install jasmine
- install karma
- configure the karma.conf.js file

[Delivers 143410115]

Signed-off-by: JULIE WANJA MUGIRA <julie.mugira@andela.com>

jm-wanja committed:
chore(testing): setup front end unit testing

- install jasmine
- install karma and its packages
- update karma.conf.js file with plugins
- change browser from phantomjs to chrome to solve connection errors

(Finishes 143410115)

Signed-off-by: JULIE WANJA MUGIRA <julie.mugira@andela.com>

nyambati commented on package.json:
```diff
@@ -33,6 +33,7 @@
     "grunt": "~1.0.1",
     "grunt-bower-task": "~0.5.0",
     "grunt-cli": "~1.2.0",
+    "karma-browserify": "^5.1.1",
```
Are we using this package anywhere???

nyambati commented on package.json:
```diff
@@ -43,14 +44,18 @@
     "passport-twitter": "~1.0.4",
+    "phantomjs": "^2.1.7",
     "pug": "^2.0.0-beta11",
+    "require.js": "^1.0.0",
```
Can you explain why we have this package in our inventory???

nyambati commented on package.json:
```diff
@@ -43,14 +44,18 @@
     "view-helpers": "~0.1.5"
   },
   "devDependencies": {
+    "browserify": "^14.3.0",
```
What are we using browserify for???

nyambati commented on package.json:
```diff
@@ -61,8 +66,19 @@
     "should": "~11.2.1",
-    "supertest": "~3.0.0"
+    "supertest": "~3.0.0",
+    "watchify": "^3.9.0"
```
What is this package used for????

nyambati commented on karma.conf.js:
```diff
@@ -0,0 +1,82 @@
+
+    // start these browsers
+    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
+    browsers: ['Chrome'],
```
Make sure you include all the browsers specified in your task.

nyambati reviewed (changes requested):
Your implemention is on point. However, it has some questionable points such dependencies that have not been used, Make the neccesary changes and resubmit your PR.

jm-wanja committed:
chore(testing): setup unit testing for front end

- remove dependencies that are not used in the program
- add PhantomJS and Firefox browsers

[Finishes #143410115]

Signed-off-by: JULIE WANJA MUGIRA <julie.mugira@andela.com>

jm-wanja committed:
chore(testing): update package.json file

- remove the karma-browserify package since it is not used in the program

[Finishes #143410115]

Signed-off-by: JULIE WANJA MUGIRA <julie.mugira@andela.com>

jm-wanja committed:
chore(testing):update package.json

- remove the require.js package
- modify .hound.yml to point to .eslintrc.json

[Finishes #143410115]

Signed-off-by: JULIE WANJA MUGIRA <julie.mugira@andela.com>
